calyx-opt/src/passes/metadata_table_gen.rs

```diff
@@ -0,0 +1,110 @@
+use crate::traversal::{Action, Named, VisResult, Visitor};
+use calyx_ir::Id;
+use calyx_utils::WithPos;
+use linked_hash_map::LinkedHashMap;
+use std::fmt;
+use std::path::PathBuf;
+
+/// Metadata stores a Map between each group name and data used in the metadata table (specified in PR #2022)
+#[derive(Default)]
+pub struct Metadata {
+    groups: LinkedHashMap<Id, (usize, PathBuf)>,
+}
+
+impl Metadata {
+    /// Create an empty metadata table
+    pub fn new() -> Self {
+        Metadata {
+            groups: LinkedHashMap::new(),
+        }
```

Comment:
Since you're deriving default, this can be replaced with a call to
`Self::default()` which does the same thing